core/server/master/src/main/java/alluxio/master/block/BlockMasterClientServiceHandler.java

```diff
@@ -120,35 +124,47 @@ public void getCapacityBytes(GetCapacityBytesPOptions options,
   @Override
   public void getUsedBytes(GetUsedBytesPOptions options,
       StreamObserver<GetUsedBytesPResponse> responseObserver) {
-    RpcUtils.call(LOG,
+    ServerRpcUtils.call(LOG,
         () -> GetUsedBytesPResponse.newBuilder().setBytes(mBlockMaster.getUsedBytes()).build(),
         "GetUsedBytes", "options=%s", responseObserver, options);
   }
 
   @Override
   public void getWorkerInfoList(GetWorkerInfoListPOptions options,
       StreamObserver<GetWorkerInfoListPResponse> responseObserver) {
-    RpcUtils.call(LOG,
-        () -> GetWorkerInfoListPResponse.newBuilder()
-            .addAllWorkerInfos(mBlockMaster.getWorkerInfoList().stream().map(GrpcUtils::toProto)
-            .collect(Collectors.toList())).build(),
-        "GetWorkerInfoList", "options=%s", responseObserver, options);
+    ServerRpcUtils.call(LOG,
+        () -> {
+          GetWorkerInfoListPResponse response = GetWorkerInfoListPResponse.newBuilder()
+              .addAllWorkerInfos(mBlockMaster.getWorkerInfoList().stream().map(GrpcUtils::toProto)
+                  .collect(Collectors.toList())).build();
+          if (response.getSerializedSize() > RPC_RESPONSE_SIZE_WARNING_THRESHOLD) {
+            LOG.warn("getWorkerInfoList response has size {}, {} WorkerInfo",
+                response.getSerializedSize(), response.getWorkerInfosCount());
+          }
```

Comment:
possibly to squeeze this into `ServerRpcUtils.call` after `T res = callable.call();` 
if we enforce  `T extending com.google.protobuf.GeneratedMessageV3` ?
then we don't need to insert this in every single case to keep your change clean

e.g. 

```
  public static <T extends GeneratedMessageV3> T callAndReturn(Logger logger, RpcCallableThrowsIOException<T> callable,
      String methodName, boolean failureOk, String description, Object... args) {
  ...
  T res = callable.call();
  if (res.getSerializedSize() > RPC_RESPONSE_SIZE_WARNING_THRESHOLD) {
    ...
  }
}

```

Reply:
Actually I thought about it and chose to make the messages more specific to each RPC, because sometimes it's hard to know where the repeated fields are. For example `BlockHeartbeatPRequest` has a list of `Metric` in the options so I have to 
do `request.getOptions().getMetricsCount()`.

I think it's kind of hard to write a generic utility method that walks the request/response recursively and report the size of  each repeated field.

Reply:
Right. We should try to avoid making this a call site change.

Reply:
maybe you can pass a string supplier to generate the warning message for each different calls. 
I will leave this up to you